### ISSUE TITLE
Replygump [gumpID] "listname[0] 

### ISSUE THREAD

**Issue by ngyoung** [jsebold666/RazorEnhanced#12]
reply gump (gumpID) (listname[0] results in an error.
"Cannot convert argument to int"

This works just fine in UOS but not in RE

**Comment by ngyoung** [comment]
It specifically doesn't like the 0 inside the brackets when doing a replygump.  Other functions referencing listname[0] seem to work fine.

**Comment by ngyoung** [comment]
Checked updates and realized I was not using the latest version.  Seems to work now but getting a different fail in my script.  I will test more and write up a new issue when I pinpoint the fail.

**Comment by ngyoung** [comment]
I am thinking that when I use the command: 
`replygump 0x554b87f3 Runes[0]`
which should be recalling off a list named Runes is automatically advancing to the next variable in the list after executing.  In UOS it does not do that.  It will stay at the same variable in the list until it gets a command like poplist or pushlist.
When running another function that accesses a list similarly like,  `promptmsg Names[0]`
you don't get the same behavior.  It will stay at the same item in the list.  I pasted the script I have been using to test the functionality.  It is a runbook copier.

```
//UOS
//Select runebook to be cloned
if not findalias 'ToBeCloned'
  headmsg 'Select Original Runebook'
  promptalias 'ToBeCloned'
endif
//Select a blank book
if not findalias 'Blank Runebook'
  headmsg 'Select Blank Runebook'
  promptalias 'Blank Runebook'
endif
//Lists
//clearlist 'Runes'
//clearlist 'Names'
if not listexists 'Runes'
  createlist 'Runes'
endif
if list 'Runes' == 0
    pushlist 'Runes' 5
    pushlist 'Runes' 11
    pushlist 'Runes' 17
    pushlist 'Runes' 23
    pushlist 'Runes' 29
    pushlist 'Runes' 35
    pushlist 'Runes' 41
    pushlist 'Runes' 47
    pushlist 'Runes' 53
    pushlist 'Runes' 59
    pushlist 'Runes' 65
    pushlist 'Runes' 71
    pushlist 'Runes' 77
    pushlist 'Runes' 83
    pushlist 'Runes' 89
    pushlist 'Runes' 95
endif
if not listexists 'Names'
  createlist 'Names'
endif
if list 'Names' == 0
  pushlist 'Names' 'Abyss E'
  pushlist 'Names' "Abyss W"
  pushlist 'Names' 'Cove W'
  pushlist 'Names' 'Deciet-Niera'
  pushlist 'Names' 'Deciet Alter'
  pushlist 'Names' 'Despise'
  pushlist 'Names' "Destard"
  pushlist 'Names' "Shame North"
  pushlist 'Names' 'Shame South'
  pushlist 'Names' "Oaks Roof"
  pushlist 'Names' 'Arch Demon'
  pushlist 'Names' 'Shame Building'
  pushlist 'Names' 'Terra Champ'
  pushlist 'Names' 'AkorOutside'
  pushlist 'Names' 'Orc East'
  pushlist 'Names' 'OrcWest'
endif
if not @findtype 0x1f14 any 'backpack'
  sysmsg 'Out of blank runes!' 25
  pause 100
  //playsound 984
  pause 100
  //playsound 987
  pause 100
  //playsound 988
  pause 100
  messagebox 'Error'  '*No Runes restock & restart*'
  stop
endif
clearjournal
for 0 to 'Runes'
  clearjournal
  if counttype 0xf7b 'any' 'backpack' < 3
    sysmsg 'out of regs' 22
    stop
  endif
  if counttype 0xf86 'any' 'backpack' < 3
    sysmsg 'out of regs' 22
    stop
  endif
  if counttype 0xf7a 'any' 'backpack' < 3
    sysmsg 'out of regs' 22
    stop
  endif
  if  mana <= 30
    useskill 'Meditation'
    while mana < maxmana
    endwhile
  endif
  pause 1500
  useobject! 'ToBeCloned'
  waitforgump 0x554b87f3 15000
  pause 900
  replygump 0x554b87f3 Runes[0]
  pause '3500'
  if injournal 'blocked' 'system'
    replay
  endif
  if @injournal 'fizzle' 'system'
    sysmsg 'your spell fizzled please restart' 23
    guildmsg 'I am being hit'
    stop
  endif
  if @injournal 'disturbed' 'system'
    sysmsg 'you were hit.. get safe and restart' 23
    guildmsg 'I am being hit'
    stop
  endif
  if @findtype 0x1f14 0 'backpack' 1 2
    autotargetobject 'found'
    cast 'Mark'
    if @injournal 'fizzle' 'system'
      sysmsg 'your spell fizzled please restart' 23
      guildmsg 'I am being hit'
      stop
    endif
    if @injournal 'disturbed' 'system'
      sysmsg 'you were hit.. get safe and restart' 23
      guildmsg 'I am being hit'
      stop
    endif
    pause 3500
    useobject! 'found'
    waitforprompt 15000
    promptmsg Names[0]
    pause 2000
    moveitem 'found' 'Blank Runebook'
    pause 2000
    if @injournal 'Names[0]' 'system'
      headmsg 'Rune added!'
      @poplist 'Names' 'front'
      @poplist 'Runes' 'front'
    endif
  else
    //playsound 984
    //pause 100
    //playsound 987
    //pause 100
    //playsound 988
    pause 100
    messagebox 'Error' '*No Runes restock & restart*'
    stop
  endif
endfor
unsetalias 'ToBeCloned'
unsetalias 'Blank Runebook'
removelist 'Runes'
removelist 'Names'
messagebox 'Error' 'Book Complete'
headmsg 'Book Complete'
```

**Comment by ngyoung** [comment]
Something still seems to not be behaving as intended.  When I run this runebook copier script that I have been testing the original error it keeps ending the script halfway through the list.  I thought maybe it was advancing the list twice but it isn't recalling to every other run location and the names stay in sync with the rune locations.  I have ran it three times with it ended at the same spot.  I am wondering if the item at the bottom of the list is getting dropped as well.  I don't have a way to see what is still in the list if I pause the script to tell.

**Comment by ngyoung** [comment]
Here is the full script that I have been using to test:
```
//UOS
// Description: Automatic Runebook Cloner.
//IT will run over and over if a location is blocked.  You will need to remove the top ## in the Rune list under Active Objects to keep going.
//Modified it to be able to stop the macro and restart.  If you fizzle script will stop the macro. Lure the spawn away then restart.  It will recall back to where you failed.
//===========================================================
//Select runebook to be cloned
if not findalias 'ToBeCloned'
  promptalias 'ToBeCloned'
endif
//Select a blank book
if not findalias 'Blank Runebook'
  promptalias 'Blank Runebook'
endif
//Lists
//clearlist 'Runes'
//clearlist 'Names'
if not listexists 'Runes'
  createlist 'Runes'
endif
if list 'Runes' == 0
    pushlist 'Runes' 5
    pushlist 'Runes' 11
    pushlist 'Runes' 17
    pushlist 'Runes' 23
    pushlist 'Runes' 29
    pushlist 'Runes' 35
    pushlist 'Runes' 41
    pushlist 'Runes' 47
    pushlist 'Runes' 53
    pushlist 'Runes' 59
    pushlist 'Runes' 65
    pushlist 'Runes' 71
    pushlist 'Runes' 77
    pushlist 'Runes' 83
    pushlist 'Runes' 89
    pushlist 'Runes' 95
endif
if not listexists 'Names'
  createlist 'Names'
endif
if list 'Names' == 0
  pushlist 'Names' 'Abyss E'
  pushlist 'Names' "Abyss W"
  pushlist 'Names' 'Cove W'
  pushlist 'Names' 'Deciet-Niera'
  pushlist 'Names' 'Deciet Alter'
  pushlist 'Names' 'Despise'
  pushlist 'Names' "Destard"
  pushlist 'Names' "Shame North"
  pushlist 'Names' 'Shame South'
  pushlist 'Names' "Oaks Roof"
  pushlist 'Names' 'Arch Demon'
  pushlist 'Names' 'Shame Building'
  pushlist 'Names' 'Terra Champ'
  pushlist 'Names' 'AkorOutside'
  pushlist 'Names' 'Orc East'
  pushlist 'Names' 'OrcWest'
endif
if not @findtype 0x1f14 any 'backpack'
  sysmsg 'Out of blank runes!' 25
  pause 100
  //playsound 984
  pause 100
  //playsound 987
  pause 100
  //playsound 988
  pause 100
  messagebox 'Error'  '*No Runes restock & restart*'
  stop
endif
clearjournal
for 0 to 'Runes'
  clearjournal
  if counttype 0xf7b 'any' 'backpack' < 3
    sysmsg 'out of regs' 22
    stop
  endif
  if counttype 0xf86 'any' 'backpack' < 3
    sysmsg 'out of regs' 22
    stop
  endif
  if counttype 0xf7a 'any' 'backpack' < 3
    sysmsg 'out of regs' 22
    stop
  endif
  if  mana <= 30
    useskill 'Meditation'
    while mana < maxmana
    endwhile
  endif
  pause 1500
  useobject! 'ToBeCloned'
  waitforgump 0x554b87f3 15000
  pause 900
  replygump 0x554b87f3 Runes[0]
  pause '3500'
  if injournal 'blocked' 'system'
    replay
  endif
  if @injournal 'fizzle' 'system'
    sysmsg 'your spell fizzled please restart' 23
    guildmsg 'I am being hit'
    stop
  endif
  if @injournal 'disturbed' 'system'
    sysmsg 'you were hit.. get safe and restart' 23
    guildmsg 'I am being hit'
    stop
  endif
  if @findtype 0x1f14 0 'backpack' 1 2
    autotargetobject 'found'
    cast 'Mark'
    if @injournal 'fizzle' 'system'
      sysmsg 'your spell fizzled please restart' 23
      guildmsg 'I am being hit'
      stop
    endif
    if @injournal 'disturbed' 'system'
      sysmsg 'you were hit.. get safe and restart' 23
      guildmsg 'I am being hit'
      stop
    endif
    pause 3500
    useobject! 'found'
    waitforprompt 15000
    promptmsg Names[0]
    pause 2000
    moveitem 'found' 'Blank Runebook'
    pause 2000
    if @injournal 'Names[0]' 'system'
      headmsg 'Rune added!'
      @poplist 'Names' 'front'
      @poplist 'Runes' 'front'
    endif
  else
    //playsound 984
    //pause 100
    //playsound 987
    //pause 100
    //playsound 988
    pause 100
    messagebox 'Error' '*No Runes restock & restart*'
    stop
  endif
endfor
unsetalias 'ToBeCloned'
unsetalias 'Blank Runebook'
removelist 'Runes'
removelist 'Names'
messagebox 'Error' 'Book Complete'
headmsg 'Book Complete'
```

**Comment by credzba** [comment]
My theory is the pop and the for using the same list is causing it to get double reduced.

**Comment by credzba** [comment]
Change replygump 0x554b87f3 Runes[0]  -> replygump 0x554b87f3 Runes[]
Remove  @poplist 'Runes' 'front'

It should work in both UOSteam and in RE UOS emulation

**Comment by ngyoung** [comment]
> Change replygump 0x554b87f3 Runes[0] -> replygump 0x554b87f3 Runes[]
> Remove @poplist 'Runes' 'front'
> 
> It should work in both UOSteam and in RE UOS emulation

I don't want it to advance the list when it executes that function.  Steam doesn't behave like that when running this script.  The only modifications that I made to it was commenting out the playsounds.  The script was written that way on purpose because in UOS using "replygump 0x554b87f3 Runes[0]" does not change the list where "replygump 0x554b87f3 Runes[]" automatically advances the list.  The way it was written confirms that the rune made it into the book with the "if @injournal 'Names[0]' 'system'" since when you add a rune to runebook it causes a system message with that runename.  Then the script knows the marked rune was successfully added.  Changing like you suggest removes the ability to for the script to remember where you last left off if you get fizzled or killed before getting the rune into the book. 

I already discovered that running with Runes[] already worked.  The very beginning of the issue was that Runes[0] would error.  Then it worked but it was stopping halfway through.  I had the same theory that it was advancing twice but it still goes to the correct recall locations.  If it was advancing twice it would be skipping over every other location.  Another theory that I had was that it was dropping variables from the back of the list as well.  This would explain why it would always stop at the halfway point in the list while still maintaining the same order at the front of the list.
